src/ec.rs

```diff
@@ -376,6 +376,32 @@ impl<S> WriterBase<S> {
     }
   }
 
+  #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+  pub fn update_cdf_4_sse2(cdf: &mut [u16; 5], val: u32) {
+    #[cfg(target_arch = "x86_64")]
+    use std::arch::x86_64::*;
+    let nsymbs = 4;
+    let rate = 3 + (nsymbs >> 1).min(2) + (cdf[nsymbs] >> 4) as usize;
+    cdf[nsymbs] += 1 - (cdf[nsymbs] >> 5);
+    #[target_feature(enable = "sse2")]
```

Comment:
Put at start of function.

Reply:
Actually the reason it has to be here is because it has to annotate the unsafe block. For it to be above the function the whole thing has to be unsafe.